paddlepaddle/paddlepaddle-model-zoo/src/test/java/ai/djl/paddlepaddle/zoo/OCRTest.java

```diff
@@ -48,68 +41,65 @@
 
         // load Character model
         Predictor<Image, String> recognizer = getRecognizer();
-        Predictor<Image, Image> rotator = getRotateClassifer();
+        Predictor<Image, Classifications> rotator = getRotateClassifer();
         for (int i = 0; i < boxes.size(); i++) {
-            Image subImg = getSubImage(img, boxes.get(i));
-            subImg = rotator.predict(subImg);
+            Image subImg = getSubImage(img, boxes.get(i).getBoundingBox());
+            if (subImg.getHeight() * 1.0 / subImg.getWidth() > 1.5) {
+                subImg = rotateImg(subImg);
+            }
+            Classifications.Classification result = rotator.predict(subImg).best();
+            if ("Rotate".equals(result.getClassName()) && result.getProbability() > 0.8) {
+                subImg = rotateImg(subImg);
+            }
             String name = recognizer.predict(subImg);
             System.out.println(name);
         }
     }
 
     @SuppressWarnings("unchecked")
-    private static List<BoundingBox> detectWords(Image img)
+    private static DetectedObjects detectWords(Image img)
             throws ModelException, IOException, TranslateException {
-        Class<List<BoundingBox>> clazz = (Class) List.class;
-        Criteria<Image, List<BoundingBox>> criteria =
-                Criteria.builder()
-                        .optEngine("PaddlePaddle")
-                        .setTypes(Image.class, clazz)
-                        .optModelUrls(
-                                "https://resources.djl.ai/test-models/paddleOCR/mobile/det_db.zip")
-                        .optTranslator(new WordsTranslator(960))
-                        .build();
+        Map<String, String> filter = new ConcurrentHashMap<>();
+        filter.put("flavor", "mobile");
 
-        try (ZooModel<Image, List<BoundingBox>> model = ModelZoo.loadModel(criteria);
-                Predictor<Image, List<BoundingBox>> predictor = model.newPredictor()) {
+        try (ZooModel<Image, DetectedObjects> model =
+                        PpModelZoo.WORD_DETECTION.loadModel(filter, null, new ProgressBar());
+                Predictor<Image, DetectedObjects> predictor = model.newPredictor()) {
             return predictor.predict(img);
         }
     }
 
     private static Predictor<Image, String> getRecognizer()
             throws MalformedModelException, ModelNotFoundException, IOException {
-        Criteria<Image, String> criteria =
-                Criteria.builder()
-                        .optEngine("PaddlePaddle")
-                        .setTypes(Image.class, String.class)
-                        .optModelUrls(
-                                "https://resources.djl.ai/test-models/paddleOCR/mobile/rec_crnn.zip")
-                        .optTranslator(new CharacterRecognitionTranslator())
-                        .build();
+        Map<String, String> filter = new ConcurrentHashMap<>();
```

Comment:
I think maybe keep Criteria is better.

Reply:
You mean keep the criteria to load from URL and use Translator in ModelZoo?